tutorials/advanced/elasticSnorkel.ipynb

```diff
@@ -0,0 +1,583 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of hits \n",
+      "8\n",
+      "Result 1\n",
+      "-------------------\n",
+      "sentence\n",
+      "Dressed in a smart black shirt and white trousers, the lawyer looked in good spirits despite a difficult fight ahead.     \n",
+      "\n",
+      "Result 2\n",
+      "-------------------\n",
+      "sentence\n",
+      "Amal, who was dressed in a smart black shirt and white trousers, looked in good spirits on Monday despite a difficult fight ahead.   \n",
+      "\n",
+      "Result 3\n",
+      "-------------------\n",
+      "sentence\n",
+      "Clad in a breton top, white trousers and a matching cap, Bob, 63, looked like he couldn't wait to get underway with the festivities.       \n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "query=\"white trousers\"\n",
+    "searchResult = eSearch.searchIndex(\"entire\",query,distance=0,size=3)\n",
+    "printResults(searchResult,\"sentence\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Positional Search\n",
+    "\n",
+    "Using a distance value for the above queries return a result that longer enforces the ordering of our query.  To search for a query where we want care about the ordering we would use the `position` keyword. This time specifying the distance defines the number of acceptable position shifts to have the query side by side, respective of order.\n",
+    "\n",
+    "The query string has to be passed in as a parameter as there individual terms. This type of query accepts any number of terms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of hits \n",
+      "2\n",
+      "Result 1\n",
+      "-------------------\n",
+      "sentence\n",
+      "It makes stops in North Carolina, Washington, D.C., Pennsylvania, Massachusetts, Minnesota, Illinois, Missouri, New York, Ohio, Florida, Georgia, Texas, California and Colorado.   \n",
+      "\n",
+      "Result 2\n",
+      "-------------------\n",
+      "sentence\n",
+      "Dozens of confused pedestrians trying to find detours around the barricades early Friday afternoon stopped on a corner of Arch Street in front of the Pennsylvania Convention Center to ask directions from Peter and Toby Christensen, an LDS couple from Colorado working as volunteers this week for the Catholic Archdiocese of Philadelphia.   \n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "searchResult = eSearch.searchIndex(\"position\",\"Pennsylvania\",\"Colorado\",distance =20)\n",
+    "printResults(searchResult,\"sentence\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Creating the Candidate Field"
```

Comment:
Can we just cut all this setup stuff and have the user feed the candidate class, e.g. `Spouse`, into the `elasticSession` constructor, and then have this stuff done automatically on init?  Is there a reason not to do this?  E.g. is this very slow?